peer/peer.go

```diff
@@ -1402,6 +1424,10 @@ out:
 		// needed.
 		rmsg, buf, err := p.readMessage(p.wireEncoding)
 		idleTimer.Stop()
+		// Send the received message to all the subscribers.
+		for _, sub := range p.subscribers {
+			sub <- rmsg
```

Comment:
Shouldn't we only do this in the case of a non-error case?

Reply:
Good point

Reply:
Fixed